src/core/Directus/Database/TableGateway/RelationalTableGateway.php

```diff
@@ -1019,7 +1019,7 @@ protected function fetchData(array $params = [])
         $meta = ArrayUtils::get($params, 'meta', false);
         $id = ArrayUtils::get($params, 'id');
         $single = ArrayUtils::get($params, 'single');
-        if (!$single && $id && count($id) == 1) {
+        if (!$single && $id && is_array($id) && count($id) == 1) {
```

Comment:
This one has been fixed already by https://github.com/directus/api/commit/bbf97f68659995bd6279f910c9501f4ba09f2f7c